tasks/main.yml

```diff
@@ -0,0 +1,97 @@
+  with_items:
+    - "{{ mattermost_hubot_install_path }}"
+    - "{{ mattermost_hubot_install_path }}/.npm"
+
+# We are using shell and sudo since npm does not respect the user under which a process should run
+- name: Install "yo" and "generator-hubot" node.js package
+  environment:
+    HOME: "{{ mattermost_hubot_install_path }}"
+    npm_config_user: "{{ mattermost_hubot_user }}"
+    npm_config_init_module: "{{ mattermost_hubot_install_path }}/.npm-init.js"
+    npm_config_userconfig: "{{ mattermost_hubot_install_path }}/.npmrc"
+    npm_config_cache: "{{ mattermost_hubot_install_path }}/.npm"
+  shell: "sudo -E -u {{ mattermost_hubot_user }} npm install {{ item }}"
+  changed_when: False
+  args:
+    chdir: "{{ mattermost_hubot_install_path }}"
+  with_items:
+    - yo
+    - generator-hubot
+
+- name: Init hubot
+  become: True
+  shell: "sudo -u {{ mattermost_hubot_user }} -E yo hubot --adapter matteruser --owner {{ mattermost_hubot_owner }} --name {{ mattermost_hubot_name }} --description '{{ mattermost_hubot_description }}'"
+  environment:
+    HOME: "{{ mattermost_hubot_install_path }}"
+    npm_config_user: "{{ mattermost_hubot_user }}"
+    npm_config_init_module: "{{ mattermost_hubot_install_path }}/.npm-init.js"
+    npm_config_userconfig: "{{ mattermost_hubot_install_path }}/.npmrc"
+    npm_config_cache: "{{ mattermost_hubot_install_path }}/.npm"
+  args:
+    chdir: "{{ mattermost_hubot_install_path }}"
+    creates: "{{ mattermost_hubot_install_path }}/scripts"
+
+- name: Delete hubot-scripts.json
+  file:
+    state: absent
+    path: "{{ mattermost_hubot_install_path }}/hubot-scripts.json"
+
+- name: Place systemd service file
+  template:
+    src: systemd.j2
+    dest: /etc/systemd/system/mattermost-hubot.service
+
+- name: Place enviroment file
+  template:
+    src: hubot_enviroment
+    dest: "/etc/default/hubot"
+    mode: 0600
+    owner: "root"
+    group: "root"
+
+- name: Install hubot plugins
+  become: True
+  become_user: "{{ mattermost_hubot_user }}"
+  environment:
+    HOME: "{{ mattermost_hubot_install_path }}"
+    npm_config_user: "{{ mattermost_hubot_user }}"
+    npm_config_group: "{{ mattermost_hubot_group }}"
+    npm_config_init_module: "{{ mattermost_hubot_install_path }}/.npm-init.js"
+    npm_config_userconfig: "{{ mattermost_hubot_install_path }}/.npmrc"
+    npm_config_cache: "{{ mattermost_hubot_install_path }}/.npm"
+  shell: "sudo -E -u {{ mattermost_hubot_user }} npm install {{ item }}"
+  changed_when: False
+  args:
+    chdir: "{{ mattermost_hubot_install_path }}"
+  with_items: "{{ mattermost_hubot_plugins }}"
+
+- name: Place plugin file service file
+  template:
+    src: external-scripts.json.j2
+    dest: "{{ mattermost_hubot_install_path }}/external-scripts.json"
+  notify:
+    - Restart systemd
+
+- name: Enable and start the systemd service
+  systemd:
+    name: mattermost-hubot
+    state: started
+    enabled: True
+    daemon_reload: True
```

Comment:
Only reload daemon when the config changes.

Reply:
fixed